align_system/utils/alignment_utils.py

```diff
@@ -53,11 +55,15 @@ def _select_min_dist_choice(self, choices, dists, misaligned=False):
             # Convert distances to probabilities
             probs = [dist/sum(dists) for dist in dists]
 
-        max_prob = max(probs)
-        max_actions = [idx for idx, p in enumerate(probs) if p == max_prob]
-        # Randomly chose one of the max probability actions
-        # TODO could add some tie breaking logic here
-        selected_choice = choices[random.choice(max_actions)]
+        if probabilistic:
+            selected_choice = np.random.choice(choices, p=probs)
+        else:
+            max_prob = max(probs)
+            max_actions = [idx for idx, p in enumerate(probs) if p == max_prob]
+            # Randomly chose one of the max probability actions
+            # TODO could add some tie breaking logic here
+            selected_choice = choices[random.choice(max_actions)]
+
```

Comment:
Not asking for any changes here, but just want to draw a little bit of attention to this being a case where having two separate functions for determining the choice (min distance or probabilistic) that gets passed in somewhere else would make it so that we don't have to keep extending the argument list here with additional flags (and what happens when we want to add another kind of flavor, and then what's the interaction between them etc.) in addition to making each individual piece easier to reason about (and write unit tests for if we decide that's necessary).

Reply:
I agree. It also felt silly that I had to add the option to every alignment function in two places (essentially)